Week3/Company_create_insert_fk.sql

```diff
@@ -0,0 +1,104 @@
+CREATE DATABASE IF NOT EXISTS Company;
+USE Company;
+
+CREATE TABLE IF NOT EXISTS Employee (
+  eID INT(4) NOT NULL,
+  eName VARCHAR(45) NOT NULL,
+  initials VARCHAR(10) NULL DEFAULT NULL,
+  dateOfBirth DATE NOT NULL,
+  position VARCHAR(20) NOT NULL,
+  salary DECIMAL(7,2) NOT NULL,
+  room VARCHAR(10) NOT NULL,
+  bossNr INT(4) NULL DEFAULT NULL,
+  dID INT(2) NOT NULL,
+  PRIMARY KEY (eID));
+
+CREATE TABLE IF NOT EXISTS Location (
+  code VARCHAR(5) NOT NULL,
+  description VARCHAR(45) NOT NULL,
+  country VARCHAR(45) NOT NULL,
+  mgrID INT(4) NULL DEFAULT NULL,
+  PRIMARY KEY (code));
+
+CREATE TABLE IF NOT EXISTS Department (
+  dID INT(2) NOT NULL,
+  dName VARCHAR(25) NOT NULL,
+  LocationCode VARCHAR(5) NOT NULL,
+  secretaryID INT(4) NULL DEFAULT NULL,
```

Comment:
Shouldn't this be a foreign key to an employee?

Reply:
If you don't create a foreign key constraint, then it will be possible to put any integer - whether it's a real employee ID or not - as this value. But because of the 'C' in ACID (consistency), the data must always follow the rules, and we definitely want it to be the case that if there is a value here, it's a real employee.